acceptance-tests/pom.xml

```diff
@@ -0,0 +1,142 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
```

Comment:
I am a bit uncomfortable having this as a subfolder rather than a proper module. This is redeclaring dependencies, versions and does not have a clear dependency on the plugin. Why did we choose to go this way? We would like to have, 

1. Plugin version at one place. Any change in the plugin should be picked up by the acceptance tests. 
2. Dependency management at one place which means we don't have to reconfigure transitive dependency in multiple places.

Reply:
Sorry, I thought I put this in the PR description, but looks like I missed it: as documented in the [Jenkins ATH wiki](https://github.com/jenkinsci/acceptance-test-harness/blob/master/docs/EXTERNAL.md):

> Acceptance test harness [is] known not to coexist well with Jenkins plugin sources. That is why it is recommended to split the sources to plugin and UI tests to individual maven modules.

Different Jenkins plugins I've checked use different maven project structures (some more weird than others) to deal with this. Check these ones for yourself (Jenkins' BlueOacean plugin + the two examples Jenkins docs provide):
https://github.com/jenkinsci/blueocean-plugin
https://github.com/jenkinsci/openstack-cloud-plugin
https://github.com/jenkinsci/kerberos-sso-plugin

For example, the BlueOcean plugin has a Maven module hierarchy, but the `acceptance-tests` module is completely separate from the other ones and doesn't share the same parent pom.

Obviously one option is to not use this framework at all, which means to implement all the things it provides ourselves, which seems completely unnecessary to me.

We're all familiar with the Maven conventions and best practices, like properly using modules, dependency management, etc. at a high/hand-wavey level, but as I've seen over and over again with almost anything to do with Jenkins, it's a world of imperfection, bad design decisions, half-baked solutions, unfamiliar conventions, etc., so I really appreciate it if we try to make sure our feedback is useful within the quite limiting bounds of what Jenkins and friends allow us to do.

So, I would like to ask for a more detailed and practical suggestion please: how would you model the maven project, given the limitations I mentioned above? I agree that "dependency management at one place which means we don't have to reconfigure transitive dependency in multiple places" is a fantastic suggestion, but can you please elaborate on how you propose we achieve that here?

Also,
> Plugin version at one place. Any change in the plugin should be picked up by the acceptance tests.

That is not an issue, as the test (see the surefire plugin configuration in the pom) always picks up the latest built `.hpi` file from the target folder in the project root, competely agnostic of the plugin version. It unfortunately doesn't _build_ the plugin before each test run, but that's why I've documented in the `readme.md` in the `acceptance-tests` module how to build the plugin before running the tests. In the CI we'll set up for these tests, the build will clone a fresh copy of the repo, build the plugin, then run the tests that would use that freshly built plugin file.

As for the repeated dependencies, although I agree it's not ideal, all dependencies here are used for the tests, or other test-related things like page objects or other helper classes, so I don't see a massive issue with, for example, the version of `commons-lang3` in the `acceptance-tests` module being different from the one used in the plugin.
Again, if there's a better approach available, I'll be happy to hear it.